tests/chainer_tests/functions_tests/math_tests/test_batch_l2_norm_squared.py

```diff
@@ -0,0 +1,72 @@
+import unittest
+
+import numpy as np
+import six
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer.functions.math.batch_l2_norm_squared import _as_two_dim
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+@testing.parameterize(
+    {'shape': (4, 3, 5)},
+    {'shape': (4, 15)},
+)
+class TestBatchL2NormSquared(unittest.TestCase):
+
+    def setUp(self):
+        self.x = np.random.uniform(-1, 1, self.shape).astype(np.float32)
+        self.gy = np.random.uniform(-1, 1, self.shape[0]).astype(np.float32)
+
+    def check_forward(self, x_data):
+        x = chainer.Variable(x_data)
+
+        y = functions.batch_l2_norm_squared(x)
+        self.assertEqual(y.data.dtype, np.float32)
+        y_data = cuda.to_cpu(y.data)
+
+        x_two_dim = _as_two_dim(self.x)
+        y_expect = np.empty(len(self.x))
+        for n in six.moves.range(len(self.x)):
+            y_expect[n] = sum(map(lambda x: x * x, x_two_dim[n]))
+
+        gradient_check.assert_allclose(y_expect, y_data)
+
+    @condition.retry(3)
+    def test_forward_cpu(self):
+        self.check_forward(self.x)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_forward_gpu(self):
+        self.check_forward(cuda.to_gpu(self.x))
+
+    def check_backward(self, x_data, y_grad):
+        x = chainer.Variable(_as_two_dim(x_data))
```

Comment:
I think `x` should not wrap reshaped `x_data` because it makes two dimensional and three dimensional case identical.